### PR TITLE
Add tls section to S3 concepts guide

### LINKED ISSUE #543: S3 concepts: Document tls usage
Add tls section to https://docs.stackable.tech/home/stable/concepts/s3

### PR DESCRIPTION
**Pull request by razvan**
Fixes #543

### CODE REVIEW

File: modules/concepts/pages/s3.adoc [hunk 1]
```diff
@@ -193,16 +203,52 @@ spec:
...
+apiVersion: v1
+kind: Secret
+metadata:
+  name: s3-certificates
+  labels:
+    secrets.stackable.tech/class: s3-certificate-class
+data:
+  ca.crt: ...
+  tls.crt: ...
+  tls.key: ...
```
Comment (sbernauer) [inline modules/concepts/pages/s3.adoc:235]: You actually only need the ca.crt. WDYT of including the SecretClass as well?

**Review by sbernauer — CHANGES_REQUESTED** [review]
LGTM otherwise, thanks!

**Comment by sbernauer** [comment]
Could we please rename the PR to something like `Add tls section to S3 concepts guide`?

File: modules/concepts/pages/s3.adoc (hunk 1 above)
Comment (razvan) [reply modules/concepts/pages/s3.adoc:253]: Updated https://github.com/stackabletech/documentation/pull/550/commits/ddb926a3c8f8f24e5b8cd5e84e2434a27b4bd7f8

File: modules/concepts/pages/s3.adoc [hunk 2]
```diff
@@ -193,16 +203,70 @@ spec:
...
```
Comment (sbernauer) [inline modules/concepts/pages/s3.adoc:253]: I think I would only include `ca.crt`, as often times I only have my company root ca, but no certificate at all. I a s a user would be "why do I need a certificate to access my S3`"
```suggestion
```

File: modules/concepts/pages/s3.adoc [hunk 3]
```diff
@@ -193,16 +203,70 @@ spec:
...
+host: test-minio
+port: 9000
 credentials:
   secretClass: s3-credentials-class
 ...
 ----
 
+== TLS
+
+TLS certificates are specified in a similar way. You will need a `SecretClass` that is referenced by the connection object. When using the `k8sSearch` backend, a `Secret` object containing the certificate/key pair is also needed.
```
Comment (sbernauer) [inline modules/concepts/pages/s3.adoc:220]:
```suggestion
TLS certificates are specified in a similar way. You will need a `SecretClass` that is referenced by the connection object. When using the `k8sSearch` backend, a `Secret` object containing the ca certificate is also needed.
```

File: modules/concepts/pages/s3.adoc [hunk 4]
```diff
@@ -193,16 +203,70 @@ spec:
...
+
+The `Secret` must contain the TLS certificate and the CA certificate. The `tls.crt` and `tls.key` are the certificate and the private key of the server. The `ca.crt` is the certificate of the Certificate Authority that signed the server's certificate. This is usually optional. The values are expected to be base64 encoded PEM files.
```
Comment (sbernauer) [inline modules/concepts/pages/s3.adoc:222]:
```suggestion
The Secret must contain the CA certificate in the form of a `ca.crt` key, which contains the Certificate Authority that signed the server's certificate. The values are expected to be base64 encoded PEM files.
```

**Review by sbernauer — COMMENTED** [review]
WDYT of this?

**Review by sbernauer — APPROVED** [review]

**Comment by sbernauer** [comment]
Thanks for working on this!

**Pull request merged**